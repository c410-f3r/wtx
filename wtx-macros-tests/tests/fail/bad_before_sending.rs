#[wtx_macros::pkg(data_format(json), id(FooId))]
mod pkg {
  #[pkg::before_sending]
  async fn before_sending(foo: i32) -> wtx::Result<()> {
    Ok(())
  }

  #[pkg::req_data]
  struct Req;

  #[pkg::res_data]
  struct Res;
}

fn main() {
}
