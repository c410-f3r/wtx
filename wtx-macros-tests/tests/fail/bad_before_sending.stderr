error: `#[pkg::before_sending]` must be an async function named `before_sending` containing any combination of `api: &mut SomeApi`, `params: &mut SomePackageParams` or `res_params: &mut SomeResponseParams`.
 --> tests/fail/bad_before_sending.rs:3:10
  |
3 |   #[pkg::before_sending]
  |          ^^^^^^^^^^^^^^

error: aborting due to 1 previous error

